lib/internal/source_map/source_map_cache.js

```diff
@@ -254,33 +275,29 @@ function appendCJSCache(obj) {
   }
 }
 
-function findSourceMap(sourceURL, isGenerated) {
+function findSourceMap(sourceURL) {
   if (RegExpPrototypeExec(kLeadingProtocol, sourceURL) === null) {
     sourceURL = pathToFileURL(sourceURL).href;
   }
   if (!SourceMap) {
     SourceMap = require('internal/source_map/source_map').SourceMap;
   }
-  let sourceMap;
-  if (isGenerated) {
-    sourceMap = generatedSourceMapCache.get(sourceURL);
-  } else {
-    sourceMap = esmSourceMapCache.get(sourceURL);
-    if (sourceMap === undefined) {
-      for (const value of cjsSourceMapCache) {
-        const filename = ObjectGetValueSafe(value, 'filename');
-        if (sourceURL === filename) {
-          sourceMap = {
-            data: ObjectGetValueSafe(value, 'data')
-          };
-        }
+  let sourceMap = esmSourceMapCache.get(sourceURL) ?? generatedSourceMapCache.get(sourceURL);
+  if (sourceMap === undefined) {
+    for (const value of cjsSourceMapCache) {
+      const filename = ObjectGetValueSafe(value, 'filename');
+      const cachedSourceURL = ObjectGetValueSafe(value, 'sourceURL');
+      if (sourceURL === filename || sourceURL === cachedSourceURL) {
+        sourceMap = {
+          data: ObjectGetValueSafe(value, 'data')
+        };
       }
     }
   }
   if (sourceMap && sourceMap.data) {
     return new SourceMap(sourceMap.data);
   }
-  return undefined;
+  return null;
```

Comment:
Isn't this a breaking change? AVA for example relies on this that it returns undefined [here](https://github.com/avajs/ava/blob/main/lib/snapshot-manager.js#L393-L394).

Reply:
Can you open a separate issue? This looks like something unintended that we can fix.

Reply:
https://github.com/nodejs/node/issues/44391